### PR TITLE
disabled the tests causing problems

### LINKED ISSUE #1125: Disable test codes causing problems such as redis
**Is your request related to a problem? Please describe.**
There are some problems, when running all tests in the Klaytn directory in local environment.
- make test
- go test ./... -v

Problems may occur when Redis and DynamoDB are not running, or depending on the version of the solidity compiler.

**Describe the solution you'd like**
I found the tests causing the problem, and disabled them all.
* add some logic that can skip the tests in local environment.
* it will be running in the circle-ci environment. 

**Environment (please complete the following information)**
- Klaytn version: v1.7.3
- go version: v1.15.7 (1.16(x), 1.17(x))
- solc version: v0.5.6

### PR DESCRIPTION
**Pull request by mckim19**
## Proposed changes
There are some problems, when running all tests in the Klaytn directory in local environment.
- make test
- go test ./... -v

Problems may occur when Redis and DynamoDB are not running, or depending on the version of the solidity compiler.

I found the tests causing the problem, and disabled them all.
I added skipLocalTest function under storage package.
And call the function at the start of the tests causing the problem.
So, the tests can be skipped when running the tests in the local environment.
 
## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

Closes #1125 

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage/database/database_test.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ var test_values = []string{"a", "1251", "\x00123\x00"}
 var testDatabases = []func() (Database, func()){newTestLDB, newTestBadgerDB, newTestMemDB, newTestDynamoS3DB}
 
 // TestDatabase_PutGet tests the basic put and get operations.
-func TestDatabase_PutGet(t *testing.T) {
```
Comment (hqjang-pepper) [inline storage/database/database_test.go:107]: Why did these four database test failed? 
I think DynamoS3DB is in `testDatabases`.
Why don't you just remove TestDynamoS3DB from `testDatabases` and try again?

**Comment by sirano11** [comment]
@mckim19 
If this PR is merged, How do i test renamed function? 
It seems to be no way to test it other than to change the name again. 

So, How about applying the logic skipping the test each environment?

File: storage/database/database_test.go (hunk 1 above)
Comment (mckim19) [reply storage/database/database_test.go:107]: Thank you for your review.
You are right. I will change to enable the rest of the databases except for the DynamoDB.

**Comment by mckim19** [comment]
>

**Comment by mckim19** [comment]
> @mckim19 If this PR is merged, How do i test renamed function? It seems to be no way to test it other than to change the name again.
> 
> So, How about applying the logic skipping the test each environment without renamed test?
> 
> https://stackoverflow.com/questions/24030059/skip-some-tests-with-go-test

**Comment by mckim19** [comment]
@sirano11 
Thank you so much for your review.
I applied the logic to skip only local tests by referring to the method you told.

**Review by aidan-kwon — APPROVED** [review]

**Comment by aidan-kwon** [comment]
@jimni1222 @sirano11 Could you review this PR?

File: storage/statedb/cache_redis_test.go [hunk 2]
```diff
@@ -86,6 +89,9 @@ func TestSubscription(t *testing.T) {
 
 // TestRedisCache tests basic operations of redis cache
 func TestRedisCache(t *testing.T) {
+	storage.SkipLocalTest(t)
+	//tests.SkipLocalTest(t)
```
Comment (jimni1222) [inline storage/statedb/cache_redis_test.go:93]:
```suggestion
```

File: storage/skip_localtest.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package storage
+
+import (
+	"os"
+	"testing"
+)
+
+func SkipLocalTest(t *testing.T) {
+	if os.Getenv("CI") == "" {
```
Comment (sirano11) [inline storage/skip_localtest.go:9]: Is this symbol added CI environment?

**Review by jimni1222 — DISMISSED** [review]

**Review by hqjang-pepper — APPROVED** [review]

File: storage/skip_localtest.go (hunk 3 above)
Comment (mckim19) [reply storage/skip_localtest.go]: Thank you for your review!
I requested the DevOps to register the CI environment variable.
And I modified the ci variable value.

**Review by sirano11 — DISMISSED** [review]

**Review by jimni1222 — DISMISSED** [review]

**Review by aidan-kwon — APPROVED** [review]

File: storage/skip_localtest.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package storage
```
Comment (jeongkyun-oh) [inline storage/skip_localtest.go:1]: @mckim19 Copyright should be added on the top of the new file.

Comment (mckim19) [reply storage/skip_localtest.go:1]: Thank you for your review. 
I wiil add the copyright.

**Review by jimni1222 — APPROVED** [review]

**Review by sirano11 — APPROVED** [review]

**Review by hqjang-pepper — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Pull request merged**